.github/workflows/continuous_integration.yml

```
@@ -0,0 +1,24 @@
+name: continuous-integration
+on: [push, pull_request]
```

```suggestion
on:
  push:
    branches:
      - main
    tags:
      - 'v*'
  pull_request:
```

If you tweak this then you won't get the CI running twice or other annoying stuff